base/uk.ac.stfc.isis.ibex.ui/src/uk/ac/stfc/isis/ibex/ui/tables/TableViewerEmptyRow.java

```diff
@@ -0,0 +1,71 @@
+package uk.ac.stfc.isis.ibex.ui.tables;
+
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.TableItem;
+import uk.ac.stfc.isis.ibex.logger.IsisLog;
+
+/**
+ * Table viewer class which will always append an empty row at the end (non-editable). This is so that
+ * copy-pasting after final (editable) line is possible.
+ */
+public class TableViewerEmptyRow extends TableViewer {
+	private static final Logger LOG = IsisLog.getLogger(TableViewerEmptyRow.class);
+	/**
+	 * Constructor for creating table using default style bits used by TableViewer.
+	 * @param parent the parent control
+	 */
+    public TableViewerEmptyRow(Composite parent) {
+        super(parent);
+    }
+    	
+    /**
+     * Constructor for creating table using style bits.
+     * @param composite the parent control
+     * @param style SWT style bits
+     */
+    public TableViewerEmptyRow(Composite composite, int style) {
+        super(composite, style);
+    }
+
+    @Override
+    public void refresh(Object element) {
+        deleteEmptyRow();
+        super.refresh(element);
```

Comment:
I'm not sure why we need to do the delete if we're only refreshing a single item? Also, if we do need to we're deleting the row but not adding it back in.